Tests/mocks/devicesettings.h

```diff
@@ -1215,6 +1228,399 @@ class FrontPanelConfig {
         return impl->getTextDisplay();
     }
 };
-
 }
 
+
+typedef uint32_t Op_t;
+class Exception : public std::exception
+{
+	public:
+		virtual const char* what() const throw()
+		{
+			return "Base Exception..";
+		}
+
+};
+class CECNoAckException : public Exception
+{
+	public:
+		virtual const char* what() const throw()
+		{
+			return "Acl not received..";
+		}
+};
+class IOException : public Exception
+{
+	public:
+		virtual const char* what() const throw()
+		{
+			return "IO Exception..";
+		}
+};
+
+class CECFrameImpl{
+	public:
+		virtual void getBuffer(const uint8_t **buf, size_t *len) const = 0;
+
+};
+class CECFrame{
+	public:
+	
+	CECFrameImpl* impl;
+	void getBuffer(const uint8_t **buf, size_t *len) const{
+		return impl->getBuffer(buf, len);
+	}
+	CECFrame(const uint8_t *buf = NULL, uint16_t len = 0){}
+
+};
+class CECBytesImpl {
+	public:
+	virtual const std::string toString() const = 0;
+};
+class CECBytes{
```

Comment:
can you help understand what are all those CECBytes classes for, do they carry any logic other than keeping a value passed in constructor and converting it to string? if that's the case you can just put some trivial implementation in here

Reply:
``So, as an example, in the base code, cecBytes toString function looks like this:

`virtual const std::string toString(void) const {
		std::stringstream stream;
		for (size_t i = 0; i < str.size(); i++) {
			stream << std::hex << (int)str[i];
		}
		return stream.str();
    };`
    
    So it pretty much is just convering it to a string.

Reply:
@npoltorapavlo Done and done.

Reply:
now you can remove CECBytesImpl, LogicalAddressImpl, PhysicalAddressImpl, LogicalAddress& getInstance(), and implement CECBytes(const uint8_t *buf, size_t len)

Reply:
Updated, though CECBytes(const uint8_t *buf, size_t len) seems to be needed, in its relation to some of the the construtors of its children.